pkl-cli/pkl-cli.gradle.kts

```diff
@@ -117,53 +248,108 @@ tasks.check {
+  exec: Provider<Exec>,
+  outputFile: File,
+  release: Boolean = enableRelease,
+) {
+  val filenameBase = outputFile.nameWithoutExtension  // accounts for `.exe`
+  val filenameSplit = filenameBase.split("-")
+
+  // `pkl-macos-<arch>` -> `macos-<arch>-<release>`
+  val releaseTag = filenameSplit.drop(1).plus(listOf(
+    if (release) "opt" else "dev"
+  )).joinToString("-")
+
+  // `macos-<arch>` -> `macos<Arch>`
+  val taskPostfix = "${filenameSplit[1]}${filenameSplit[2].capitalized()}"
+
+  // `macos<Arch>` -> `macos<Arch>DistZip
+  val zipTaskName = "${taskPostfix}DistZip"  // `macosAmd64DistZip`
+  val tarTaskName = "${taskPostfix}DistTar"  // `macosAmd64DistTar`
+
+  distributions.create(taskPostfix) {
+    distributionBaseName = "pkl-cli"
+    distributionClassifier = releaseTag
+    contents {
+      from(layout.projectDirectory.dir("src/dist"), exec.get().outputs.files)
+      include("*")
+      rename {
+        if (!it.startsWith("pkl-")) it else "pkl"
+      }
+    }
+  }
+  exec.get().apply {
+    finalizedBy(zipTaskName, tarTaskName)
+  }
+}
+
 fun Exec.configureExecutable(isEnabled: Boolean, outputFile: File, extraArgs: List<String> = listOf()) {
-  enabled = isEnabled
-  dependsOn(":installGraalVm")
+  enabled = isEnabled && isNativeBuildEnabled
+  dependsOn(":installGraalVm", releasePrep)
 
   inputs.files(sourceSets.main.map { it.output })
   inputs.files(configurations.runtimeClasspath)
   outputs.file(outputFile)
 
   workingDir = outputFile.parentFile
   executable = "${buildInfo.graalVm.baseDir}/bin/native-image"
+  outputs.cacheIf { true }
 
   // JARs to exclude from the class path for the native-image build.
   val exclusions =
     if (buildInfo.graalVm.isGraal22) emptyList()
-    else listOf(libs.truffleApi, libs.graalSdk).map { it.get().module.name }
+    else nativeImageExclusions.map { it.get().module.name }
   // https://www.graalvm.org/22.0/reference-manual/native-image/Options/
   argumentProviders.add(CommandLineArgumentProvider {
     listOf(
+        "--strict-image-heap"
         // currently gives a deprecation warning, but we've been told 
         // that the "initialize everything at build time" *CLI* option is likely here to stay
-        "--initialize-at-build-time="
+        ,"--initialize-at-build-time="
         ,"--no-fallback"
         ,"-H:IncludeResources=org/pkl/core/stdlib/.*\\.pkl"
         ,"-H:IncludeResources=org/jline/utils/.*"
         ,"-H:IncludeResources=org/pkl/commons/cli/commands/IncludedCARoots.pem"
         //,"-H:IncludeResources=org/pkl/core/Release.properties"
         ,"-H:IncludeResourceBundles=org.pkl.core.errorMessages"
-        ,"--macro:truffle"
+        ,"--macro:truffle-svm"
+        ,"-H:Module=$module"
         ,"-H:Class=org.pkl.cli.Main"
         ,"-H:Name=${outputFile.name}"
         //,"--native-image-info"
-        //,"-Dpolyglot.image-build-time.PreinitializeContexts=pkl"
+        ,"-Dpolyglot.image-build-time.PreinitializeContexts=pkl"
```

Comment:
This was re-enabled and necessary support was added to the Pkl language engine implementation.